### ISSUE TITLE
Wouldn't be good to use intel mkl for CPU provider?

### ISSUE THREAD

**Issue by snarb** [alex-weaver/TensorSharp#1]

**Comment by alex-weaver** [comment]
Yes, it probably would :) The main bottleneck in most usages is likely to be the BLAS routines, and it should be relatively straightforward to amend the code to use MKL instead of OpenBLAS for these (See src/TensorSharp/Cpu/OpenBlasNative.cs)

I didn't want to make MKL a requirement for running ops on the CPU because downloading MKL requires each user to register, and I think it's good to keep the friction low for getting started. Of course I have no objection to introducing MKL as an *optional* backend optimization. Pull requests are always welcome :)
